tests/socket/socket_tests.cpp

```diff
@@ -33,6 +34,23 @@ CATCH_REGISTER_LISTENER(_watchdog)
 
 #define MULTIPLE_ATTACH_PROGRAM_COUNT 3
 
+thread_local bool _is_main_thread = false;
+
+struct test_failure : std::exception
+{
+    test_failure(const std::string& message) : message(message) {}
+    std::string message;
+};
+
+#define SAFE_REQUIRE(x)                                               \
```

Comment:
Should we make this in a test lib for common usage across all tests?

Reply:
I think we cannot use this for the tests which are also used with fault injection tests. For all the kernel tests though, agree this can be a common code.